File: benchmarks/nsieve.js
// Integer loops + array mutation; Sieve of Eratosthenes (n = 2^14).
function nsieve(n) {
    const flags = new Array(n);
    for (let i = 0; i < n; i++) {
        flags[i] = 1;
    }
    let count = 0;
    for (let p = 2; p < n; p++) {
        if (flags[p] === 1) {
            count++;
            for (let k = p + p; k < n; k += p) {
                flags[k] = 0;
            }
        }
    }
    return count;
}

console.log(nsieve(1 << 14));

File: benchmarks/nsieve.lua
-- Integer loops + array mutation; Sieve of Eratosthenes (n = 2^14).
local function nsieve(n)
    local flags = {}
    for i = 0, n - 1 do
        flags[i] = 1
    end
    local count = 0
    for p = 2, n - 1 do
        if flags[p] == 1 then
            count = count + 1
            for k = p + p, n - 1, p do
                flags[k] = 0
            end
        end
    end
    return count
end

print(nsieve(1 << 14))
